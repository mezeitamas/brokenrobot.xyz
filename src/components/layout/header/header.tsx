import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { InternalLink } from '../../internal-link/internal-link';

type HeaderProps = {
    title: string;
};

const Header: FunctionComponent<HeaderProps> = ({ title }): ReactElement => {
    return (
        <header className="mb-12 mt-6 flex flex-col gap-4 leading-8">
            <nav className="flex flex-col justify-between gap-4 whitespace-nowrap sm:flex-row">
                <div className="text-2xl font-bold text-gray-900">
                    <InternalLink to="/">{title}</InternalLink>
                </div>

                <div className="flex flex-nowrap gap-6 text-gray-600 sm:gap-10">
                    <InternalLink to="/">Home</InternalLink>
                    <InternalLink to="/blog">Blog</InternalLink>
                    <InternalLink to="/about-me">About me</InternalLink>
                </div>
            </nav>

            <hr />
        </header>
    );
};

export { Header };

export type { HeaderProps };
