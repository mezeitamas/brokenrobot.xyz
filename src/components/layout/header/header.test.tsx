import React from 'react';

import { create } from 'react-test-renderer';

import { Header } from './header';
import type { HeaderProps } from './header';

describe('<Header />', () => {
    test('should render', () => {
        // Given
        const props: HeaderProps = {
            title: 'Broken Robot'
        };

        // When
        const component = create(<Header {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
