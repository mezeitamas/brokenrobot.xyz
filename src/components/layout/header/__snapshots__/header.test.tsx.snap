// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
<header
  className="mb-12 mt-6 flex flex-col gap-4 leading-8"
>
  <nav
    className="flex flex-col justify-between gap-4 whitespace-nowrap sm:flex-row"
  >
    <div
      className="text-2xl font-bold text-gray-900"
    >
      <span
        className="hover:underline"
      >
        <a
          href="/"
        >
          Broken Robot
        </a>
      </span>
    </div>
    <div
      className="flex flex-nowrap gap-6 text-gray-600 sm:gap-10"
    >
      <span
        className="hover:underline"
      >
        <a
          href="/"
        >
          Home
        </a>
      </span>
      <span
        className="hover:underline"
      >
        <a
          href="/blog"
        >
          Blog
        </a>
      </span>
      <span
        className="hover:underline"
      >
        <a
          href="/about-me"
        >
          About me
        </a>
      </span>
    </div>
  </nav>
  <hr />
</header>
`;
